### DISCUSSION TITLE
How to let lightning store hyperparameters in checkpoint when working with Hydra configs?

### DISCUSSION THREAD

**Discussion by mariomeissner** [Lightning-Universe/lightning-transformers#234] [Q&A]
In my personal hydra lightning setup, I am passing certain instantiated objects as parameters to my model (e.g. a loss function). It's convenient because I can say `_target_: torch.nn.CrossentropyLoss` in the config file. 

The problem is that lightning will fail to store the hyperparameters in the checkpoint file because it is not pickable. How can I make this work? I ask here because I assume that this problem had to be somehow overcome here too, but a quick look at the source code didn't give me the answer yet.

Interestingly, inside the lightning checkpoint saving code, [there is some logic](https://github.com/PyTorchLightning/pytorch-lightning/blob/a14783ea8c4475c59b5bbaff81f592b3e8d90481/pytorch_lightning/trainer/connectors/checkpoint_connector.py#L371) trying to identify if we are working with OmegaConf containers, but it's still not straightforward for me to adapt my code to this situation.

**Comment by Borda** [comment]
@SeanNaren, mind haveing a look? :)

**Reply by mariomeissner** [reply]
I personally solved this one by storing object references as classpath strings instead. Not ideal, as it's fancier to pass a hydra-initialized object rather than a string...

**Reply by mariomeissner** [reply]
Either way, it seems wanted to move over to Lightning CLI, is that still a plan in the works? ^^
